cmd/edit.go

```
@@ -0,0 +1,145 @@
+	if err != nil {
+		return fmt.Errorf("can't create Kibana client: %w", err)
+	}
+
+	kibanaVersion, err := kibanaClient.Version()
+	if err != nil {
+		return fmt.Errorf("can't get Kibana status information: %w", err)
+	}
+
+	if kibanaVersion.IsSnapshot() {
+		message := fmt.Sprintf("editing dashboards from a SNAPSHOT version of Kibana (%s) is discouraged. It could lead to invalid dashboards (for example if they use features that are reverted or modified before the final release)", kibanaVersion.Version())
+		if !allowSnapshot {
+			return fmt.Errorf("%s. --%s flag can be used to ignore this error", message, cobraext.AllowSnapshotFlagName)
+		}
+		fmt.Printf("Warning: %s\n", message)
+	}
+
+	if len(dashboardIDs) == 0 {
+		dashboardIDs, err = promptDashboardIDs(kibanaClient)
+		if err != nil {
+			return fmt.Errorf("prompt for dashboard selection failed: %w", err)
+		}
+
+		if len(dashboardIDs) == 0 {
+			fmt.Println("No dashboards were found in Kibana.")
+			return nil
+		}
+	}
+
+	var updatedDashboardIDs []string
+	var errMsgs []string
+	for _, dashboardID := range dashboardIDs {
+		err = kibanaClient.SetManagedSavedObject("dashboard", dashboardID, false)
+		if err != nil {
+			errMsgs = append(errMsgs, err.Error())
+		} else {
+			updatedDashboardIDs = append(updatedDashboardIDs, dashboardID)
+		}
+	}
+
+	if len(errMsgs) > 0 {
+		cmd.Println(fmt.Sprintf("\nFailed to make the following dashboards editable in Kibana:\n%s", strings.Join(errMsgs, "\n")))
+	}
+	if len(updatedDashboardIDs) > 0 {
+		urls, err := dashboardURLs(*kibanaClient, updatedDashboardIDs)
+		if err != nil {
+			cmd.Println(fmt.Sprintf("\nFailed to retrieve dashboard URLS: %s", err.Error()))
+			cmd.Println(fmt.Sprintf("The following dashboards are now editable in Kibana:\n%s", strings.Join(updatedDashboardIDs, "\n")))
+		} else {
+			cmd.Println(fmt.Sprintf("\nThe following dashboards are now editable in Kibana:%s", urls))
+		}
+	}
+	return nil
```

If there were errors we probably don't want to exit successfully, we should return an error on these cases.

---

Ah, good point!